docs/variants/asus_kgpe_d16/setup.md

```diff
@@ -0,0 +1,214 @@
+
+flashrom -w firmware.bin -p linux_spi:dev=/dev/spidev1.0,spispeed=16000
+
+# select flash <-> board
+echo "1" > /sys/class/gpio/gpio400/value
+```
+
+ RTE header J10 pin | qspimux J101 pin
+:------------------:|:-------------------:
+ 1 (GPIO400)        | 9 (MUX_SEL)
+ 2 (GPIO401)        | 4 (IO3_HOLD_PROG)
+
+![](/images/qspimux_pin_header.png)
+
+![](/images/qspimux_kgpe.jpeg)
+
+> qspimux schematics: https://github.com/felixheld/qspimux/blob/master/qspimux/qspimux.pdf
+
+#### Serial
+
+Asus KGPE-D16 has a DB9 serial connector for debug UART. Connect it to the RTE
+via RS232 DB9 cable.
+
+![](/images/kgpe_rear_panel.png)
+
+#### TPM
+
+Asus KGPE-D16 has a LPC header for TPM or debugging. Since there is no other
+option to connect a TPM, we use the header for that purpose.
+
+Example connection of PC Engines TPM 20pin module (tpm1a) to TPM header on
+KGPE-D16:
+
+![](/images/kgpe_tpm_header.png)
+
+| tpm1a         | Asus KGPE-D16        |
+|:-------------:|:--------------------:|
+| PCLK pin 1    | TPM pin 1  (PCICLK)  |
+| LAD0 pin 3    | TPM pin 11 (LAD0)    |
+| LAD1 pin 5    | TPM pin 10 (LAD1)    |
+| LAD2 pin 7    | TPM pin 8  (LAD2)    |
+| LAD3 pin 9    | TPM pin 7  (LAD3)    |
+| FRAME# pin 11 | TPM pin 3  (FRAME)   |
+| RESET# pin 13 | TPM pin 5  (PCIRST#) |
+| GND pin 17    | TPM pin 2  (GND)     |
+| V3 pin 18     | TPM pin 9  (+3V)     |
+| SERIRQ pin 19 | TPM pin 16 (SERIRQ)  |
+
+> For PC Engines TPM pinout refer to https://www.pcengines.ch/schema/tpm1a.pdf
+
+#### Power management
+
+Asus KGPE-D16 provides a front panel pin header (PANEL1):
+
+![](/images/kgpe_front_panel.png)
+
+| RTE header J11 pin     | Asus KGPE-D16             |
+|:----------------------:|:-------------------------:|
+| 8 (OC buffer output)   | PANEL1 pin 11 (PWR)       |
+| 9 (OC buffer output)   | PANEL1 pin 17 (RESET)     |
+
+Additionally connect one of the PANEL1 ground pins to one of the ground pins on
+RTE.
+
+### MANUAL
+
+For more details about KGPE-D16 refer to the manual:
+https://dlcdnets.asus.com/pub/ASUS/mb/SocketG34(1944)/KGPE-D16/Menual_QVL/E8847_KGPE-D16.pdf
+
+### Theory of operation
+
+#### Flashing
+
+There are two RTEs connected to two Asus KGPE-D16 boards:
+
+* 192.168.4.236
+* 192.168.4.111
+
+Both boards are equipped with the qspimux and a replaced flash chip (Winbond
+W25Q64FV 8MB). The flash chip is attached to the qspimux with an adapter
+available with qspimux. Flashing may be done via the commands shown earlier or
+the flash script on present in the root directory of the RTE:
+
+```shell
+./flash.sh coreboot.rom
+```
+
+The script will automatically power off the board and update flash chip
+contents using flashrom. After the operation is finished the routing of SPI
+signals is restored to the platform.
+
+#### Power management
+
+RTE is able to control the power of the board with the `rte_ctrl` application
+or with the web GUI available at RTE's IP address port 8000. Possible command are:
```

Comment:
@miczyg1 it is worh to mentioned more references to RTE, there are some information on Github as well in documentation. Source code to `rte_ctrl` tool also can be pointed e..g if someone would like to fix/extend the tool.

Reply:
I believe we do not have the source for rte_ctrl... We may later replace it with a bash script with equivalent functionality

Reply:
really? https://github.com/3mdeb/RteCtrl

Reply:
This is the bash for testing: https://github.com/3mdeb/RteCtrl/blob/master/test/rteCtrl.sh

Reply:
Really. This is the RteCtrl service code wit the GUI. `rte_ctrl` is the old command line tool which I could not find source for.

Reply:
Added more RTE references. In terms of rte_ctrl. I guess we can leave it for now, but maybe we should swich to using RteCtrl exclusively.

Reply:
We should switch to something what code we have.